src/mds/CDir.cc

```diff
@@ -1857,6 +1863,11 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
   mark_complete();
   state_clear(STATE_FETCHING);
 
+  if (scrub_infop && scrub_infop->need_scrub_local) {
+    scrub_infop->need_scrub_local = false;
+    scrub_local();
```

Comment:
We're again losing our chance to report back the results of the local scrub by invoking it as part of the fetch, which will cause trouble for validate_disk_state...